### DISCUSSION TITLE
Use Paketo Buildpack instead of Dockerfile

### DISCUSSION THREAD

**Discussion by tschuehly** [basecamp/kamal#481] [Q&A] [answered]
I would like to try using Kamal for a Spring Boot Application.

With Spring Boot you don't need to write your own Dockerfile and instead use the Gradle/Maven Plugin to build a Image: https://docs.spring.io/spring-boot/docs/current/gradle-plugin/reference/htmlsingle/#build-image

This uses Paketo Buildpack to create a optimized Docker Image.

Can I run ./gradlew bootBuildImage to build the image with Kamal somehow?

**Comment by nlsrchtr** [comment]
Hi @tschuehly,

I'm using the Paketo Buildpacks for a Rails application as well. In my use case, I use a CI pipeline build the image and use kamal for making the deployment, using the "-P" option to disable local builds.

I'm pretty sure you can run Paketo Buildpacks locally for building your image.

**Comment by gregjotau** [comment, marked as answer]
@tschuehly 

`alias ,d="./gradlew bootBuildImage && VERSION=latest kamal deploy -P"`

works as a deploy alias with this gradle config: 

```
// creates smaller docker images, buildpacks
tasks.named<org.springframework.boot.gradle.tasks.bundling.BootBuildImage>("bootBuildImage") {
    imageName.set("username/servicename:latest")
    publish.set(true)
    docker {
        publishRegistry {
            username = "username"
            password = System.getenv("DOCKER_REGISTRY_TOKEN")
        }
    }

    environment.set(
        mapOf(
            "BP_IMAGE_LABELS" to "service=\"servicename\"",
            "BP_JVM_VERSION" to "23"
        )
    )
}
```

See https://github.com/basecamp/kamal/issues/497 also

**Comment by nickhammond** [comment]
@tschuehly Feel free to add any feedback you have to this [buildpack PR for Kamal](https://github.com/basecamp/kamal/pull/916).
